client/client-lib/src/lib.rs

```diff
@@ -1415,16 +1416,39 @@ impl Client<GatewayClientConfig> {
     /// Wait for a lightning preimage gateway has purchased to be decrypted by
     /// the federation
     pub async fn await_preimage_decryption(&self, outpoint: OutPoint) -> Result<Preimage> {
-        Ok(self
-            .context
-            .api
-            .await_output_outcome::<OutputOutcome>(
-                outpoint,
-                Duration::from_secs(10),
-                &self.context.decoders,
+        let deadline = Instant::now().add(Duration::from_secs(30));
+
+        let poll = || async {
```

Comment:
Poll looks like a variant of retry. Is our existing [retry utility ](https://github.com/fedimint/fedimint/blob/master/gateway/ln-gateway/src/utils.rs#L16) applicable for this?

Reply:
It has to be deadline based, not attempt-s based, so I don't think so.

Reply:
I don't know if it's worth adding one for it either. Maybe if we converted more code to use deadlines it would make sense.